terraform-beyond-layer0/example-3/modules/guestbook_service/Dockerrun.aws.json

```diff
@@ -0,0 +1,35 @@
+{
```

Comment:
I think I know what you're saying but not sure I completely agree. 

> the module for the guestbook service you are importing wouldn't exist in the same repo/directory tree

This might be the case if you intend for your modules to be reusable components across different projects but that doesn't have to be the case right? You can just use modules to group a set of resources or provide a layer of abstraction. In the case of example 3, the modules directory exists outside the the different environments. I.e.,

```
example-3
  |-env-dev
  |-env-staging
  |-modules
      |- guestbook_service
```

I would argue that the purpose here is to avoid duplicate configuration per environment (dev, staging). In theory, the env-dev and env-staging can import the guestbook_service module from  anywhere. Placing the guestbook_service in the same repo/directory tree is more for convenience but I don't know if we can conclusively say that it is a pattern users shouldn't use?